packages/suite/src/hooks/wallet/coinmarket/form/useCoinmarketSatsSwitcher.ts

```diff
@@ -0,0 +1,67 @@
+import { Account } from '@suite-common/wallet-types';
+import { amountToSatoshi, formatAmount } from '@suite-common/wallet-utils';
+import { useDidUpdate } from '@trezor/react-utils';
+import { FORM_CRYPTO_INPUT, FORM_FIAT_INPUT } from 'src/constants/wallet/coinmarket/form';
+import { useBitcoinAmountUnit } from 'src/hooks/wallet/useBitcoinAmountUnit';
+import { Network } from 'src/types/wallet';
+import { coinmarketGetRoundedFiatAmount } from 'src/utils/wallet/coinmarket/coinmarketUtils';
+
+interface UseCoinmarketSatsSwitcherProps {
+    account: Account;
+    methods: any;
+    quoteCryptoAmount: string | undefined;
+    quoteFiatAmount: string | undefined;
+    network: Network | null;
+    setIsSubmittingHelper: (value: boolean) => void;
+}
+
+/**
+ * Hook for switching between crypto and fiat amount in coinmarket Sell and Buy form
+ */
+export const useCoinmarketSatsSwitcher = ({
+    account,
+    methods,
+    quoteCryptoAmount,
+    quoteFiatAmount,
+    network,
+    setIsSubmittingHelper,
+}: UseCoinmarketSatsSwitcherProps) => {
+    const { setValue, getValues } = methods;
+    const { shouldSendInSats } = useBitcoinAmountUnit(account.symbol);
+    const networkDecimals = network?.decimals ?? 8;
```

Comment:
Consider to import the `8` OR Consider to extract as DefaultCoinmarketNetworkDecimals